snapcraft/parts/lifecycle.py

```diff
@@ -164,7 +164,7 @@ def _run_command(  # noqa PLR0913 # pylint: disable=too-many-branches, too-many-
         project.parts,
         work_dir=work_dir,
         assets_dir=assets_dir,
-        base=project.get_effective_base(),
+        base=project.get_effective_base(allow_return_devel=project.type == "base"),
```

Comment:
If I understand this correctly, `craft-parts` needs to know the base is `devel` only when building the devel base snap.  Why is that?

Reply:
`base` snap does not has `base` defined, which created an issue for the checking later.  It actually can be anything but None.

Reply:
Sounds good, do you mind adding a comment about that in the code?

Reply:
Sure